### PR TITLE
chore: refactor all bridge calls to codegen and api to use a facade

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
Migrate all API and Codegen bridge requests to use facades.

#### Issue #, if available
N/A

#### Description of how you validated changes
Unit + [e2e tests](https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/13101/workflows/b9c1480b-2bfa-451a-b6f7-e26b8751847d)

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli-core/API.md [hunk 1]
```diff
@@ -232,6 +232,10 @@ export class AngularConfigNotFoundError extends Error {
 
 // @public (undocumented)
 export class ApiCategoryFacade {
+    // (undocumented)
+    static addGraphQLAuthorizationMode(context: $TSContext, authType: string): Promise<void>;
+    // (undocumented)
+    static generateContainersArtifacts(context: $TSContext, resource: $TSAny): Promise<$TSObject>;
     // (undocumented)
```
Comment (sobolk) [inline packages/amplify-cli-core/API.md:239]: does this and below depend in any way on `@aws-amplify/graphql-transformer-core` or `graphql-transformer-core` ?

Comment (alharris-at) [reply packages/amplify-cli-core/API.md:239]: No new dependencies, there's an implicit dep here on the api category, but nothing where we're adding the transformer-core deps directly.

File: packages/amplify-cli-core/src/plugin-facade/api-category-facade.ts [hunk 2]
```diff
@@ -36,4 +37,27 @@ export class ApiCategoryFacade {
...
+  }
+
+  /**
+   * Generates an ECS Stack and metadata for the given resource object.
+   */
+  static async generateContainersArtifacts(
+    context: $TSContext,
+    resource: $TSAny,
+  ): Promise<$TSObject> {
+    return context.amplify.invokePluginMethod(context, 'api', undefined, 'generateContainersArtifacts', [context, resource]);
```
Comment (sobolk) [inline packages/amplify-cli-core/src/plugin-facade/api-category-facade.ts:61]:
```suggestion
    return context.amplify.invokePluginMethod(context, API_CATEGORY_NAME, undefined, 'generateContainersArtifacts', [context, resource]);
```

File: packages/amplify-cli-core/API.md (hunk 1 above)
Comment (sobolk) [reply packages/amplify-cli-core/API.md:239]: I think that in long term we should find better place to host plugin Facades. Perhaps separate package. Core should be leaf dependency and just bring utilities.

**Review by sobolk — DISMISSED** [review]

**Comment by alharris-at** [comment]
Closing, as discussed w/ Kamil offline, and we probably can get rid of these facades altogether.

**Pull request closed without merging**